auction-server/src/auction/service/verification.rs

```diff
@@ -1245,7 +1252,19 @@ impl Service<Svm> {
             .await;
         match simulation {
             Ok(simulation) => {
-                if simulation.value.err.is_some() {
+                if let Some(transaction_error) = simulation.value.err {
+                    if let TransactionError::InstructionError(_, instruction_error) =
+                        transaction_error
+                    {
+                        if instruction_error
+                            == InstructionError::Custom(ErrorCode::InsufficientUserFunds.into())
+                        {
+                            // This path only works as long as none of the other accepted programs use this error number, which is currently true since express relay is the only program in the whitelist that uses anchor error codes
+                            // TODO: Also check the instruction index here so we are sure it's coming from the swap instruction
```

Comment:
wouldn't this TODO resolve the above comment? since ER doesn't make any CPIs to anything except system & token program?

Reply:
YES! let me clarify it